### PR TITLE
Instance type M6 and Volume gp3, AMI update to Amazon Linux 2 2.0.20230628.

### PR DESCRIPTION
**Pull request by matafc**
## Changes
- Updates the volume type for the bastion and container instances to gp3, from gp2.
- Defines m6 type instances as EBS optimized, per https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSOptimized.html . This will allow us to update the instances to m6i 
- Sets the MaxBatchSize for AutoscalingRollingUpdate to 20. This should reduce `resource contention` or `API Rate Limit exceeded` type errors when applying Cloudformation changes.
- Also including the latest AMI update so our instances will remain compliant. Details at the section below

This has been tested on a local instance of Barcelona with the sre-sandbox AWS account.

### How to roll out

We can update the instance type and volume type at the same time. Both changes require a `bcn district apply `, and Cloudformation change set execution. These steps will allow you to only perform it once.

- Merge and confirm that this PR's changes has been deployed to Barcelona.
- `bcn api patch /districts/:district_name '{ "cluster_instance_type": "m6i.large"}'` # replace `:district_name`
- `bcn district apply :district_name`
- Review and execute the Cloudformation changeset that was produced.

## AMI Update

This PR will also update the AMIs of the container instances as described in this [guide](https://www.notion.so/How-to-Update-AMI-for-komoju-district-6980c4e127774d9791d44d9ad51b0321):

[Release notes](https://docs.aws.amazon.com/AL2/latest/relnotes/relnotes-20230629.html)

[Amazon ECS-optimized AMIs - Amazon Elastic Container Service](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-optimized_AMI.html)

ECS AMI SSM path: https://ap-northeast-1.console.aws.amazon.com/systems-manager/parameters/aws/service/ecs/optimized-ami/amazon-linux-2/recommended/image_id/description?region=ap-northeast-1
https://ap-northeast-1.console.aws.amazon.com/systems-manager/parameters/aws/service/ecs/optimized-ami/amazon-linux-2/recommended/description?region=ap-northeast-1

And it also updates the AMI for the bastion image.

Bastion AMI SSM path: https://ap-northeast-1.console.aws.amazon.com/systems-manager/parameters/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2/description?region=ap-northeast-1

### CODE REVIEW

**Comment by degikko** [comment]
@essa can you help us review this PR, please?

**Review by essa — APPROVED** [review]
LGTM 👍 
Thank you!

File: lib/barcelona/network/autoscaling_builder.rb [hunk 1]
```diff
@@ -5,28 +5,28 @@ class AutoscalingBuilder < CloudFormation::Builder
...
+        "ap-southeast-2"      => "ami-037bc2c139c7ae160",
+        "ca-central-1"      => "ami-0519bf5ddc298485d",
+        "ap-south-1"      => "ami-025fa2a3e27b6e58a",
+        "sa-east-1"      => "ami-0f544759009d3c50b",
       }
 
       def ebs_optimized_by_default?
         # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSOptimized.html
-        !!(instance_type =~ /\A(a1|c4|c5.?|d2|f1|g3.?|h1|i3|m4|m5.?|p2|p3(dn)?|r4|r5.?|t3|u-.*|x1.?|z1d)\..*\z/)
+        !!(instance_type =~ /\A(a1|c4|c5.?|d2|f1|g3.?|h1|i3|m4|m5.?|m6()?|p2|p3(dn)?|r4|r5.?|t3|u-.*|x1.?|z1d)\..*\z/)
```
Comment (davidsiaw) [inline lib/barcelona/network/autoscaling_builder.rb:29]: @matafc is there a reason we use empty parens

`...|m6()?|...`

Doesnt seem like we are backreferencing anything (usually the reason we do this)

Comment (matafc) [reply lib/barcelona/network/autoscaling_builder.rb:29]: Good catch, that is not right at all. I meant to update that to match the m6 types. I'll make a new PR to address this.

**Pull request merged**